bilby/gw/prior.py

```diff
@@ -502,17 +505,43 @@ def __init__(self, a_prior=Uniform(0, 1), z_prior=Uniform(-1, 1),
         chi_min = min(a_prior.maximum * z_prior.minimum,
                       a_prior.minimum * z_prior.maximum)
         chi_max = a_prior.maximum * z_prior.maximum
-        xx = np.linspace(chi_min, chi_max, 800)
-        a_prior_minimum = a_prior.minimum
-        if a_prior_minimum == 0:
-            a_prior_minimum += 1e-32
-        aas = np.linspace(a_prior_minimum, a_prior.maximum, 1000)
-        yy = [np.trapz(np.nan_to_num(a_prior.prob(aas) / aas *
-                                     z_prior.prob(x / aas)), aas) for x in xx]
-        super(AlignedSpin, self).__init__(xx=xx, yy=yy, name=name,
-                                          latex_label=latex_label, unit=unit,
-                                          boundary=boundary, minimum=minimum,
-                                          maximum=maximum)
+        if self._is_simple_aligned_prior:
+            xx = np.linspace(chi_min, chi_max, 100000)
+            yy = - np.log(np.abs(xx) / a_prior.maximum) / (2 * a_prior.maximum)
+        else:
+
+            def integrand(aa, chi):
+                """
+                The integrand for the aligned spin (chi) probability density
+                after performing the integral over spin orientation using a
+                delta function identity.
+                """
+                return a_prior.prob(aa) * z_prior.prob(chi / aa) / aa
+
+            xx = np.linspace(chi_min, chi_max, 10000)
+            yy = [
+                quad(integrand, a_prior.minimum, a_prior.maximum, chi)[0]
+                for chi in xx
```

Comment:
The old version of this had `1e-32` add if the minimum was zero, was there a reason to remove this?

Reply:
I think it is because the integrand has a `1 / spin_mag` term, and then trapz would complain. Since quad is used now, this is not a problem.